static/app/actionCreators/account.tsx

```diff
@@ -1,18 +1,31 @@
 import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
 import {Client} from 'app/api';
 import ConfigStore from 'app/stores/configStore';
-import {Identity, User} from 'app/types';
+import {User, UserIdentityConfig} from 'app/types';
 
-export async function disconnectIdentity(identity: Identity) {
+export async function disconnectIdentity(
+  identity: UserIdentityConfig,
+  onSuccess: {(): void}
+) {
   const api = new Client();
 
   try {
-    await api.requestPromise(`/users/me/social-identities/${identity.id}/`, {
-      method: 'DELETE',
-    });
-    addSuccessMessage(`Disconnected ${identity.providerLabel}`);
+    await api.requestPromise(
+      `/users/me/user-identities/${identity.category}/${identity.id}/`,
+      {
+        method: 'DELETE',
+      }
+    );
   } catch {
     addErrorMessage('Error disconnecting identity');
+    return;
+  }
+
+  addSuccessMessage(`Disconnected ${identity.provider.name}`);
+  try {
+    onSuccess();
+  } catch {
+    addErrorMessage('Error: Please refresh');
```

Comment:
What are you try catching the onSuccess for?

Reply:
Also should be translated and maybe the string should just be `There was a problem, try refreshing`

Reply:
This was actually throwing an error at one point in dev because I was a dope and messed up the scope of `this`. So... it could happen again? 😅

Mainly I just wanted it out of the upper try block, so that some error on the page refresh doesn't falsely signal that the deletion operation failed. But you're right, it's probably safe to just lose this second catch block entirely.

Reply:
I think it's probably safe to drop this :)